pkg/lib/synthesize/resource_getters.go

```diff
@@ -0,0 +1,313 @@
+package synthesize
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiregistrationv1beta1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+const (
+	CSVKind                = "ClusterServiceVersion"
+	CRDKind                = "CustomResourceDefinition"
+	SecretKind             = "Secret"
+	ConfigMapKind          = "ConfigMap"
+	ClusterRoleKind        = "ClusterRole"
+	ClusterRoleBindingKind = "ClusterRoleBinding"
+	ServiceAccountKind     = "ServiceAccount"
+	ServiceKind            = "Service"
+	RoleKind               = "Role"
+	RoleBindingKind        = "RoleBinding"
+	PrometheusRuleKind     = "PrometheusRule"
+	ServiceMonitorKind     = "ServiceMonitor"
+	DeploymentKind         = "Deployment"
+	APIServiceKind         = "APIService"
+)
+
+// registerGVK registers the GVK of the kube resource we use in the CSV, explicitly including all versions.
+func registerGVK(scheme *runtime.Scheme) {
+	scheme.AddKnownTypes(operatorsv1alpha1.SchemeGroupVersion, &operatorsv1alpha1.ClusterServiceVersion{})
+
+	scheme.AddKnownTypes(apiextensionsv1.SchemeGroupVersion, &apiextensionsv1.CustomResourceDefinition{})
+	scheme.AddKnownTypes(apiextensionsv1beta1.SchemeGroupVersion, &apiextensionsv1beta1.CustomResourceDefinition{})
+
+	scheme.AddKnownTypes(corev1.SchemeGroupVersion,
+		&corev1.ServiceAccount{},
+		&corev1.ConfigMap{},
+		&corev1.Secret{})
+
+	scheme.AddKnownTypes(rbacv1.SchemeGroupVersion,
+		&rbacv1.Role{},
+		&rbacv1.RoleBinding{},
+		&rbacv1.ClusterRole{},
+		&rbacv1.ClusterRoleBinding{})
+
+	scheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.Deployment{})
+
+	scheme.AddKnownTypes(apiregistrationv1.SchemeGroupVersion, &apiregistrationv1.APIService{})
+	scheme.AddKnownTypes(apiregistrationv1beta1.SchemeGroupVersion, &apiregistrationv1beta1.APIService{})
+}
+
+type GroupVersionKindName struct {
```

Comment:
Use [`DefinitionKey`](https://github.com/operator-framework/operator-registry/blob/master/pkg/registry/types.go#L35) instead?

Reply:
nice find, didn't see this.